Sources/HTTP/Serializer/RequestSerializer.swift

```diff
@@ -0,0 +1,72 @@
+import Transport
+
+public final class RequestSerializer<Stream: WriteableStream>: Serializer {
+    typealias StreamType = Stream
+    let stream: Stream
+    public init(_ stream: Stream) {
+        self.stream = stream
+    }
+    
+    public func serialize(_ request: Request) throws {
+        let message = try serialize(request as Message)
+        
+        let methodBytes: Bytes
+        switch request.method {
+        case .connect:
+            methodBytes = [.C, .O, .N, .N, .E, .C, .T]
+        case .delete:
+            methodBytes = [.D, .E, .L, .E, .T, .E]
+        case .get:
+            methodBytes = [.G, .E, .T]
+        case .head:
+            methodBytes = [.H, .E, .A, .D]
+        case .options:
+            methodBytes = [.O, .P, .T, .I, .O, .N, .S]
+        case .put:
+            methodBytes = [.P, .U, .T]
+        case .patch:
+            methodBytes = [.P, .A, .T, .C, .H]
+        case .post:
+            methodBytes = [.P, .O, .S, .T]
+        case .trace:
+            methodBytes = [.T, .R, .A, .C, .E]
+        case .other(let string):
+            methodBytes = string.makeBytes()
+        }
+        
+        var length = methodBytes.count
+            + 1 //space
+            + request.uri.path.characters.count
+        if let query = request.uri.query {
+            length += 1 + query.characters.count
+        }
+        length += 11 // _HTTP/1.1\r\n
```

Comment:
even though 11 is based on something, I'd still rather call have a global constant fileprivate or something like `versionLength` ... should this be dependent on the actual version? ie, is it possible for there ever to be 1.12?